dev/tasks/build/builder.js

```diff
@@ -0,0 +1,238 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* jshint node: true */
+
+'use strict';
+
+module.exports = Builder;
+
+/**
+ * A CKEditor 5 release builder.
+ *
+ * @class Builder
+ */
+function Builder( target ) {
+	/**
+	 * The target directory where to create the build.
+	 *
+	 * **Warning**: if existing, this directory will be deleted before processing.
+	 *
+	 * @property {String}
+	 */
+	this.target = target || 'build';
+
+	/**
+	 * The temporary directory to use for build processing.
+	 *
+	 * **Warning**: if existing, this directory will be deleted before processing.
+	 *
+	 * @property {String}
+	 */
+	this.tmp = 'build-tmp';
+
+	/**
+	 * The list of tasks to be executed by the `build()` method. Each entry is an Array containing the name of the
+	 * method inside `tasks` to execute and the message to show to the end user when executing it.
+	 *
+	 * @property {Array}
+	 */
+	this.taskList = [
+		[ 'cleanUp', 'Cleaning the "' + this.target + '" directory...' ],
+		[ 'copyToTmp', 'Copying source files for manipulation...' ],
+		[ 'removeAmdNamespace', 'AMD cleanup...' ],
+		[ 'optimize', 'Creating the optimized code...' ],
+		[ 'cleanUpTmp', 'Removing the "' + this.tmp + '" directory...' ]
+	];
+}
+
+Builder.prototype = {
+	/**
+	 * Builds a CKEditor release based on the current development code.
+	 *
+	 * @param {Function} [callback] Function to be called when build finishes. It receives `false` on error.
+	 */
+	build: function( callback ) {
+		var that = this;
+		var stepCounter = 0;
+
+		// Before starting, run the initial checkups.
+		if ( !this.checkUp() ) {
+			console.error( 'Build operation aborted.' );
+			callback( false );
+
+			return;
+		}
+
+		console.log( 'Building CKEditor into the "' + this.target + '" directory:')
```

Comment:
Fixed with cccf119.

Damn WebStorm that doesn't work with our pre-commit hook. They fixed it some good time ago but still it didn't land into any of their releases.